modules/statistics/RTPStatsCollector.js

```diff
@@ -722,6 +742,16 @@ StatsCollector.prototype.processStatsReport = function() {
         }
     }
 
+    this.eventEmitter.emit(
```

Comment:
Why was this moved? It's close to where byteSentStats is done being calculated?

Honestly, I've always had a hard time understanding the 400 line block of multiple nested ifs that is processStatsReport. I'm glad you tried breaking this apart.

Reply:
Emitting BYTE_SENT_STATS depends on local variable and I left it in that method, while extracting: this._processAndEmitReport();, as it is used in two places. 
I can also move the emit BYTE_SENT_STATS inside _processAndEmitReport, by passing byteSentStats to the _processAndEmitReport method. Does this sound ok?

Reply:
No, it's okay where it is probably. I just couldn't see the intent of the move without some context.